console-api/proto/resources.proto

```diff
@@ -37,8 +37,10 @@ message Resource {
     common.MetaId metadata = 2;
     // The resources's concrete rust type.
     string concrete_type = 3;
+    // The location in code where the resource was created.
+    common.Location location = 4;
     // The kind of resource (e.g timer, mutex)
-    Kind kind = 4;
+    Kind kind = 5;
```

Comment:
i'd really prefer to make the protobuf and console-subscriber changes in a separate branch from the console change that adds the resources UI, so that the git history is easier to read. if it's not too much work, would you mind moving the proto/subscriber changes to a separate PR and having this branch depend on that? thanks!